lib/core/sessions.js

```diff
@@ -708,6 +708,10 @@ function applySession(session, command, options) {
     return new MongoError('Cannot use a session that has ended');
   }
 
+  if (options && options.writeConcern && options.writeConcern.w === 0) {
```

Comment:
I think it would make sense to put a comment here, something like
`// SPEC-1019: silently ignore explicit session with unacknowledged write for backwards compatibility`